### PR TITLE
bpo-44042: sqlite3 begin transaction optimisations

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue44042

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Changes are not user visible, so I'm adding the _skip news_ label.

File: Modules/_sqlite/cursor.c [hunk 1]
```diff
@@ -416,15 +415,15 @@ begin_transaction(pysqlite_Connection *self)
...
 
-error:
-    if (PyErr_Occurred()) {
-        return -1;
-    }
     return 0;
+
+error:
+    _pysqlite_seterror(self->db);
+    return -1;
```
Comment (erlend-aasland) [inline Modules/_sqlite/cursor.c]: @serhiy-storchaka, @berkerpeksag: I'm a little bit unsure about this change. What if `Py_BEGIN_ALLOW_THREADS` or `Py_END_ALLOW_THREADS` fail?

**Comment by erlend-aasland** [comment]
cc. @corona10

**Comment by corona10** [comment]
@erlend-aasland 

This is PR looks good to be labeled as `skip news`

File: Modules/_sqlite/cursor.c (hunk 1 above)
Comment (erlend-aasland) [reply Modules/_sqlite/cursor.c]: Never mind. I've reverted 393368f89cd6f4a02693d5e9e863bbbf9eb19498 for now, so the error handling is unchanged.

**Review by pablogsal — APPROVED** [review]

**Pull request merged**